prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -370,17 +372,14 @@ def answer_binary_market(
                 )
                 self.save_answer_to_long_term_memory(prediction)
 
-        final_answer = (
-            self.generate_final_decision(
-                question, scenarios_with_probs, created_time=created_time
-            )
-            if scenarios_with_probs
-            else None
+            if len(scenarios_with_probs) < len(all_scenarios) / 2:
+                raise ValueError(
```

Comment:
Before it would error out only if all subpredictions will fail, but having a threshold of 50% feels better